### ISSUE TITLE
[Bug][Medium] Tax-loss harvesting caps losses at a hardcoded $3,000 "IRS limit" - wrong country, wrong currency

### ISSUE THREAD

**Issue by hansika-singh** [omroy07/AI-Money-Mentor#488]
## Description
`utils/auto_rebalancer.py`'s `get_tax_harvesting_opportunities()` caps every
harvestable capital loss at a flat `3000`, labeled as the "IRS limit":

```python
'harvestable_loss': min(abs(pnl), 3000),  # IRS limit
'suggestion': f'Consider selling {h["symbol"]} to realize ₹{min(abs(pnl), 3000):,.2f} loss'
```

This app is India-focused (all amounts elsewhere are in ₹/INR, per the UI
and other modules like `loan_planner.py`, `tax.py`, `tax_filing.py`). The
IRS is the **US** tax authority, and its $3,000/year capital-loss deduction
limit is denominated in **US Dollars**, not Indian Rupees. Applying this
constant directly to ₹ amounts is a ~₹249,000 currency mismatch (at
~₹83/USD) and also imports a US-specific tax rule into an Indian tax
context, where capital-loss set-off/carry-forward rules are entirely
different (Income Tax Act, no flat rupee cap like this).

## Demonstrated impact
A holding with a genuine ₹50,000 loss:

```python
harvestable_loss = min(abs(pnl), 3000)  # => ₹3,000
```

The app tells the user they can only harvest **₹3,000** of a **₹50,000**
loss  understating the harvestable amount by 94%, and the "IRS limit"
label is factually wrong for an Indian user.

## Steps to reproduce
1. Call `AutoRebalancer.get_tax_harvesting_opportunities()` (or trigger it
   from the rebalancer feature) with a holding that has `buy_price` well
   above `current_price` (a loss of, say, ₹50,000+).
2. Observe `harvestable_loss` is capped at ₹3,000 regardless of the actual
   loss size, and the suggestion text quotes the same wrong figure.

## Suggested fix
Remove the flat `3000` cap entirely (or replace it with the correct Indian
rule) - the harvestable loss should just be the realized loss itself:

```python
'harvestable_loss': round(abs(pnl), 2),
'suggestion': f'Consider selling {h["symbol"]} to realize ₹{abs(pnl):,.2f} loss'
```

If a cap is genuinely intended (e.g. to model India's short-term vs
long-term capital loss set-off rules), it needs to be re-derived from
Indian tax law and clearly labeled  not copied from the unrelated US
$3,000 IRS rule.

**Comment by hansika-singh** [comment]
Hii @maintainers
I would like to reslove this issue as a part of SSoC26

**Comment by arcgod-design** [comment]
Hi @omroy07! I am participating in SSoC '26 and would love to work on this issue.

My approach:
- Replace the hardcoded `3000` IRS limit in `utils/auto_rebalancer.py`'s `get_tax_harvesting_opportunities()` with a configurable locale-aware parameter
- For India-focused usage, remove the US-specific capital loss cap and implement India's Income Tax Act rules (set-off against capital gains, no flat rupee cap)
- Add a `country` or `locale` config option so the logic can be adapted to other jurisdictions in the future
- Update unit tests to cover Indian tax rules and add a regression test for the wrong-currency bug

Could you please assign this issue to me? I'm ready to start immediately!
